rfcs/056-prismic-etl-pipeline/README.md

```diff
@@ -0,0 +1,91 @@
+# RFC 056: Prismic to Elasticsearch ETL pipeline
+
+**Status:** Draft
+
+**Last updated:** 23/2/2023
+
+## Context
+In order to make our editorial content - including stories, comics, exhibitions and events - more discoverable, we want to be able to search it via an API as described in [RFC 055](https://github.com/wellcomecollection/docs/tree/main/rfcs/055-prismic-api).
+
+While Prismic does provide some search functionality of its own (which we have been using for initial versions of unified search), we want more control and fewer limitations. To achieve this we want to use Elasticsearch, as we do for our other search services. As such, we need to get data from Prismic into Elasticsearch: this RFC will propose the mechanism by which we achieve that.
+
+#### Desiderata
+- Changes (including additions and edits but possibly not deletions - see [questions](#questions-and-potential-issues)) in Prismic are reflected promptly in the Elasticsearch index
+- Full reindexes are easy, quick and cheap to perform
+- Changes to the data mapping (and the index mapping) can be made easily by any developer
```

Comment:
I would expand this slightly, to cover further back in the process – it should be possible for any developer who's updating the Prismic model to update this data mapping *if necessary*, i.e.

- it should be possible to change the Prismic model definitions and leave this pipeline as-is if you're changing unmapped fields
- it should be possible to change the Prismic model definitions and coordinate your change with this pipeline

Reply:
Ah yes, good point (fortunately don't think it makes much difference to implementation?)

Reply:
> fortunately don't think it makes much difference to implementation?

yeah, I think it's more a scenario to bear in mind rather than imposing big requirements

(I suppose the only way it might affect implementation is if we can create a consistent approach to doing this sort of task, but that could just as easily be "existing Prismic tasks copy the new way" as "the new way copies the existing Prismic tasks")